internal/gps/vcs_source.go

```diff
@@ -40,6 +41,10 @@ func (bs *baseVCSSource) upstreamURL() string {
 	return bs.repo.Remote()
 }
 
+func (bs *baseVCSSource) commitInfo(ctx context.Context, r Revision) (*vcs.CommitInfo, error) {
```

Comment:
Is it appropriate to return this type?

We could return `(commit, author, message string, timestamp time.Time, err error)` to avoid tangling up too tightly with an external project.

Another option is to make this a more targeted method, like `commitFor(ctx context.Context, specifier string) (Revision, error)`.